libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,106 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
+use error_stack::Report;
+use image::DynamicImage;
+use pdfium_render::prelude::*;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+/// Function to read the pdf
 ///
-/// ```rust
-/// use chonky::add;
+/// # Errors
+/// Will return `Err` if `filename` does not exist or the user does not have
+/// permission to read it.
+pub fn load_pdf<'a>(
+    pdfium: &'a Pdfium,
+    file_path: &str,
+) -> Result<PdfDocument<'a>, Report<PdfiumError>> {
+    let pdf = pdfium.load_pdf_from_file(file_path, None)?;
+    Ok(pdf)
```

Comment:
There is no need to create a temporary file. However, the `Ok` and `?` is required here as it's converting `PdfiumError` to `Report<PdfiumError>`.
I'd go with either of these:
```suggestion
    pdfium.load_pdf_from_file(file_path, None).map_err(Report::new)
```
or 
```suggestion
    Ok(pdfium.load_pdf_from_file(file_path, None)?)
```

Either works, the first one is more explicit (and probably requires `rustfmt` to run 😅) and I probably like this more. Whatever suits you best!

Reply:
I will most likely stick to the first one also because its clear in its intention.